app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -61,6 +62,11 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
 
         val intent = SafeIntent(intent)
 
+        if (Settings.getInstance(this).shouldShowFirstrun() &&
+                !Performance.processIntentIfPerformanceTest(intent, this)){
+            components.appStore.dispatch(AppAction.ShowFirstRun)
```

Comment:
thought (non-blocking): this is just a summary of my understanding of this pattern. This is a timeline of the events:
```
appStore.dispatch(ShowFirstRun) // dispatches to appStore dispatcher thread
lifecycle.addObserver(navigator)

navigator.start() -> subscribe() // subscribes to appStore in onStart

navigator.subscribe$collect // changes screens on state change
```

One thing I don't understand is why the navigator is guaranteed to receive the event. Since the dispatch occurs on a background thread, could the state update occur before we subscribe in `onStart`? I'm not familiar with the flow API: maybe it's propagating the initial state to the subscribers and that's why it's safe? Or perhaps it's because the framework (probably) doesn't yield the UI thread until after `onStart` so we have time to subscribe before we collect?

@jonalmeida Do you know if this is safe?

---

A second thought is if there is anything in `HomeActivity.onCreate/onStop/etc.` that shouldn't be called because we've navigated to the onboarding screen. I don't think I see anything. It may be added in the future but I'm not sure how to protect against that.

Reply:
>  It may be added in the future but I'm not sure how to protect against that.

Does the intent that we send go directly to the `MainActivity` instead of the `IntentReceiverActivity` first? We could protect against this by processing the intent at the other activity and fire the action immediately before the `MainActivity` is even created.

Just a thought.